src/execCmd.ts

```diff
@@ -0,0 +1,210 @@
+    }
+  }
+  return result;
+};
+
+const getExitCodeError = (expectedCode: number, actualCode: number, cmd: string) => {
+  return Error(`Unexpected exit code for command: ${cmd}. Expected: ${expectedCode} Actual: ${actualCode}`);
+};
+
+/**
+ * Synchronously execute a command with the provided options in a child process.
+ *
+ * Option defaults:
+ *    1. `cwd` = process.cwd()
+ *    2. `timeout` = 300000 (5 minutes)
+ *    3. `env` = process.env
+ *    4. `silent` = true (child process output not written to the console)
+ *
+ * Other defaults:
+ *
+ *    @see www.npmjs.com/package/shelljs#execcommand--options--callback
+ *    @see www.nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback
+ *
+ * @param cmd The command string to be executed by a child process.
+ * @param options The options used to run the command.
+ * @returns The child process exit code, stdout, stderr, cmd run time, and the parsed JSON if `--json` param present.
+ */
+export const execCmd = (cmd: string, options: ExecCmdOptions = {}): ExecCmdResult => {
+  const debug = Debug('testkit:execCmd');
+
+  // Ensure we run synchronously
+  if (options.async) {
+    throw new Error('execCmd must be run synchronously.  Use execCmdAsync to run asynchronously.');
+  }
+
+  const cmdOptions = Object.assign({}, DEFAULT_SHELL_OPTIONS, options);
+
+  debug(`Running cmd: ${cmd}`);
+  debug(`Cmd options: ${inspect(cmdOptions)}`);
+
+  const result: ExecCmdResult = {
+    shellOutput: '' as ShellString,
+    execCmdDuration: Duration.seconds(0),
+  };
+
+  // Execute the command in a synchronous child process
+  const startTime = process.hrtime();
+  result.shellOutput = shelljs.exec(cmd, cmdOptions) as ShellString;
+  result.execCmdDuration = hrtimeToMillisDuration(process.hrtime(startTime));
+  debug(`Command completed with exit code: ${result.shellOutput.code}`);
+
+  if (isNumber(cmdOptions.ensureExitCode) && result.shellOutput.code !== cmdOptions.ensureExitCode) {
+    throw getExitCodeError(cmdOptions.ensureExitCode, result.shellOutput.code, cmd);
+  }
+
+  return addJsonOutput(cmd, result);
+};
+
+/**
+ * Asynchronously execute a command with the provided options in a child process.
+ *
+ * Option defaults:
+ *    1. `cwd` = process.cwd()
+ *    2. `timeout` = 300000 (5 minutes)
+ *    3. `env` = process.env
+ *    4. `silent` = true (child process output not written to the console)
+ *
+ * Other defaults:
+ *
+ *    @see www.npmjs.com/package/shelljs#execcommand--options--callback
+ *    @see www.nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback
+ *
+ * @param cmd The command string to be executed by a child process.
+ * @param options The options used to run the command.
+ * @returns The child process exit code, stdout, stderr, cmd run time, and the parsed JSON if `--json` param present.
+ */
+export const execCmdAsync = async (cmd: string, options: ExecCmdOptions = {}): Promise<ExecCmdResult> => {
```

Comment:
What would we lose if we only had the async command?  Then you'd always use it in the exact same way, without caring whether it's async or not or having two nearly identical methods?

Reply:
I think we definitely want both.  It's more flexible especially if you don't want to have to think about promises or async/await in your tests.  We've learned from the core lib that only providing async options is too limiting.  NodeJS API provides similar options.